dpe/src/x509.rs

```diff
@@ -1135,6 +1408,118 @@ impl X509CertWriter<'_> {
 
         Ok(bytes_written)
     }
+
+    /// Encode a certification request info
+    ///
+    /// Returns number of bytes written to `scratch`
+    ///
+    /// CertificationRequestInfo ::= SEQUENCE {
+    /// version       INTEGER { v1(0) } (v1,...),
+    /// subject       Name,
+    /// subjectPKInfo SubjectPublicKeyInfo{{ PKInfoAlgorithms }},
+    /// attributes    [0] Attributes{{ CRIAttributes }}} }
+    ///
+    /// # Arguments
+    ///
+    /// * `pubkey` - ECDSA Public key.
+    /// * `subject_name` - The subject name RDN struct to encode.
+    /// * `measurements` - DPE measurement data.
+    pub fn encode_certification_request_info(
+        &mut self,
+        pub_key: &EcdsaPub,
+        subject_name: &Name,
+        measurements: &MeasurementData,
+    ) -> Result<usize, DpeErrorCode> {
+        let cert_req_info_size = Self::get_certification_request_info_size(
+            subject_name,
+            pub_key,
+            measurements,
+            /*tagged=*/ false,
+        )?;
+
+        // CertificationRequestInfo Sequence
+        let mut bytes_written = self.encode_tag_field(Self::SEQUENCE_TAG)?;
+        bytes_written += self.encode_size_field(cert_req_info_size)?;
+
+        // version
+        bytes_written += self.encode_version()?;
+
+        // subject
+        bytes_written += self.encode_rdn(subject_name)?;
+
+        // subjectPublicKeyInfo
+        bytes_written += self.encode_ecdsa_subject_pubkey_info(pub_key)?;
+
+        // extensions
+        bytes_written += self.encode_extensions(measurements)?;
```

Comment:
I think there needs to be a wrapper struct around the extensions. There is an attribute "Extension Request" defined in RFC 2985 that should wrap these extensions.

Reply:
Done